### ISSUE TITLE
Clone file and results file names contain path if based on argv[0]

### ISSUE THREAD

**Issue by galexv** [ALPSCore/ALPSCore#223]
Running `/path/to/bin/alps_mc_program /path/to/param/params.ini` set defaults for  `checkpoint` and `outputfile` parameters to `params.clone.h5` and `params.out.h5` as it should.

Running `/path/to/bin/alps_mc_program --some-args=x` set defaults for  `checkpoint` and `outputfile` parameters to  `/path/to/bin/alps_mc_program.clone.h5` and `/path/to/bin/alps_mc_program.out.h5`, which is wrong. The checkpoint and result file names should contain only the base name, not the full path.

**Comment by galexv** [comment]
The problem is slightly different: the defaults for `checkpoint` and `outputfile` *always* contain the full path; however, when we restore from the checkpoint, the parameters have the defaults they had when the checkpoint file was created. This is a bit confusing, and can lead to unexpected results, as illustrated below:

```sh
$ cd /path/to/my_bin_dir
$ mkdir my_run_dir
$ cd my_run_dir
$ ../alps_mc_prog ../params.ini
```
Checkpoint is saved to  `../params.clone.h5`, that is, `/path/to/my_bin_dir/params.clone.h5`. Now continue:
```sh
$ cd /path/to/my_bin_dir
$ ./alps_mc_prog params.clone.h5
```
*Oops!* It tries to read the checkpoint from `../params.clone.h5`, that is, from the parent directory of `my_bin_dir` &mdash; it does not exists!
Okay, we can override this:
```sh
$ cd /path/to/my_bin_dir
$ ./alps_mc_prog params.clone.h5 --checkpoint=params.clone.h5
```
It now runs, but again *oops!* Now the output file `params.out.h5` is created in the parent directory!

Besides, having the checkpoint file by default at the same place as the executable (when all the parameters are passed via the command line) is also inconvenient/unexpected: i would rather expect it to create checkpoints in the current directory.

To summarize: it's not technically a bug, but needs improvement: namely, let's strip the leading paths from the filenames.
